src/tir/usmp/algo/hill_climb.cc

```diff
@@ -155,8 +156,8 @@ class HillClimbAllocator : public GreedyBase {
   void collect_neighbor_lists(const BufferInfoNode* buf,
                               std::vector<const BufferInfoNode*>* first_level,
                               std::vector<const BufferInfoNode*>* second_level, const TPos& _pos) {
-    std::unordered_map<int, const BufferInfoNode*> first_level_set;
-    std::unordered_map<int, const BufferInfoNode*> second_level_set;
+    std::map<int, const BufferInfoNode*> first_level_set;
```

Comment:
This would be costly (in terms of runtime) just to maintain stability.
I think we should explore a seperated ordered structure to maintain this instead (additionally).

Reply:
amended